src/core/built-in/rules/root-files.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const fs = require('fs');
+const { promisify } = require('util');
+
+const Rule = require('../../plugins/rule');
+const docs = require('../../../utils/docs');
+
+const readDirAsync = promisify(fs.readdir);
+
+class RootFiles extends Rule {
+  constructor(context) {
+    super(context);
+
+    if (!this.context.options['required'] && !this.context.options['blacklist']) {
+      throw new Error(`The rule must have at least one option, "required" or "blacklist"`);
+    }
+
+    if (this.context.options['required'] && !Array.isArray(this.context.options['required'])) {
+      throw new Error(`Wrong "required" argument, an array is expected`);
+    }
+
+    if (this.context.options['blacklist'] && !Array.isArray(this.context.options['blacklist'])) {
+      throw new Error(`Wrong "blacklist" argument, an array is expected`);
+    }
+  }
+
+  async run(sandbox) {
+    const requiredFiles = this.context.options['required'];
+    const blacklistedFiles = this.context.options['blacklist'];
+
+    const files = await this._readRootDirectory();
+
+    if (requiredFiles) {
+      for (let index = 0; index < requiredFiles.length; index++) {
+        if (!files.includes(requiredFiles[index])) {
+          sandbox.report({
+            message: `Your directory root is missing the required file ${requiredFiles[index]}`
+          });
+          return;
```

Comment:
See above about messaging about all files, instead of just the first missing one